p2p/utils.py

```diff
@@ -47,3 +54,33 @@ def get_process_pool_executor() -> ProcessPoolExecutor:
     else:
         cpu_count = os_cpu_count - 1
     return ProcessPoolExecutor(cpu_count)
+
+
+class ThroughputTracker:
+    """
+    Tracks throughput using an exponential moving average.
+    https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average
+    """
+    def __init__(self, default_throughput: Union[int, float], smoothing_factor: float) -> None:
```

Comment:
nitpick: Is there a reason for using `Union[int, float]` over simply sticking to `float` entirely for this API?

Reply:
I guess it's just a claim that both will be fine. No objection to dropping to float, though.